### PR TITLE
Roll out `ScriptInterface`

### PR DESCRIPTION
**Pull request by jngrad**
Description of changes:
- convert Cython files to Python files using the `ScriptInterface` framework:
   - particle handle/list/slice classes
   - system class and globals class
   - comfixed, galilei, analysis, polymer functions
   - code_info, cuda_init, version modules
- make `ScriptInterface` conversion error messages clearer by printing the contents of Python containers, so as to provide a viable alternative to `utils.check_type_or_throw_except()`
- remove timestamp from `config.hpp` and sort features alphabetically to improve `ccache` efficiency
- API change: class attributes expecting 3 boolean values no longer accept integer values
   - affected properties: `system.periodicity`, `particle.fix`, `particle.rotation`
   - it is no longer possible to set those properties with e.g. `[1, 1, 1]` or `[0, 0, 1]`
- declutter `src/core` and `src/script_interface` by grouping similar features into submodules
   - `src/core/*.{h,c}pp` files are now for the most part declaring or using global variables (snake case filenames), while `src/core/*/*.{h,c}pp` mostly contain class definitions (Pascal case filenames)
   - `src/script_interface/*.{h,c}pp` files now all contain the `ScriptInterface` classes and helper functions, while `src/script_interface/*/*.{h,c}pp` contains the consumer code that bridges the core classes to the Python classes

### CODE REVIEW

**Comment by jngrad** [comment]
This PR needs to be merged before the coding day. It currently blocks the non-bonded interaction refactoring (coding day task) and the CMake `FindPython` refactoring (already deployed on the `walberla` branch).

File: doc/sphinx/particles.rst [hunk 1]
```diff
@@ -654,7 +654,7 @@ Lattice-Boltzmann swimmers
 
     import espressomd
     system = espressomd.System(box_l=[1, 1, 1])
-    system.part.add(pos=[2, 0, 0], rotation=[1, 1, 1], swimming={
+    system.part.add(pos=[2, 0, 0], rotation=[True, True, True], swimming={
```
Comment (christophlohrmann) [inline doc/sphinx/particles.rst:657]: ``3 * [True]`` ?

File: src/python/espressomd/polymer.py [hunk 2]
```diff
@@ -75,21 +46,21 @@ def linear_polymer_positions(**kwargs):
...
+        Angle in radians between adjacent bonds within a polymer.
+        Must be in the range :math:`[0, \\pi]`. If omitted, random
+        angles are drawn.
+    start_positions : (N, 3) array_like :obj:`float`, optional
         If set, this vector defines the start positions for the polymers, i.e.,
         the position of each polymer's first monomer bead.
-        Here, a numpy array of shape (n_polymers, 3) is expected.
+        Here, a numpy array of shape ``(n_polymers, 3)`` is expected.
     min_distance : :obj:`float`, optional
-        Minimum distance between all generated positions. Defaults to 0
```
Comment (christophlohrmann) [inline src/python/espressomd/polymer.py:85]: I liked the default statement

File: testsuite/python/engine_lb.py [hunk 3]
```diff
@@ -84,13 +84,6 @@ def tearDown(self):
         self.system.actors.clear()
         self.system.thermostat.turn_off()
 
-    def test_conflicting_parameters(self):
```
Comment (christophlohrmann) [inline testsuite/python/engine_lb.py:87]: why is this test gone?

Comment (jngrad) [reply testsuite/python/engine_lb.py:87]: It has been improved and moved there:
https://github.com/espressomd/espresso/blob/606247de636ea8e9921fab6b3cd883727f8be709/testsuite/python/particle.py#L371-L375

File: src/python/espressomd/polymer.py (hunk 2 above)
Comment (jngrad) [reply src/python/espressomd/polymer.py:85]: Isn't it redundant, considering Sphinx displays `min_distance=0.` in the function parameters list?

File: doc/sphinx/particles.rst (hunk 1 above)
Comment (jngrad) [reply doc/sphinx/particles.rst:657]: I once tried to change these and box_l to the `3 * [x]` syntax, but I got push back from the core team. Some people prefer the `[x] * 3` syntax, others prefer `[x, x, x]`. I guess it's a matter of taste.

File: src/python/espressomd/polymer.py (hunk 2 above)
Comment (christophlohrmann) [reply src/python/espressomd/polymer.py:85]: didn't see that, nvm then

**Review by christophlohrmann — DISMISSED** [review]

**Review by reinaual — DISMISSED** [review]

**Review by reinaual — DISMISSED** [review]

**Review by reinaual — APPROVED** [review]

**Pull request merged**